src/AutoRest.CSharp/LowLevel/Output/DpgOutputLibraryBuilder.cs

```diff
@@ -209,10 +209,26 @@ private static InputOperation UpdateOperation(InputOperation operation, string c
             }
             operationsMap.Add(operation, updatedOperation);
 
+            if (!updatedOperation.Parameters.Any(p => p.IsEndpoint || (p.Location == RequestLocation.Uri && p.Kind == InputOperationParameterKind.Client)))
+            {
+                updatedOperation = updatedOperation with
+                {
+                    Parameters = AddEndpointParameter(updatedOperation.Parameters)
+                };
+            }
+
             return updatedOperation;
 
         }
 
+        private static IReadOnlyList<InputParameter> AddEndpointParameter(IReadOnlyList<InputParameter> operationParameters)
+        {
+            return new List<InputParameter>(operationParameters)
+            {
+                new InputParameter(KnownParameters.Endpoint.Name, "endpoint", $"{KnownParameters.Endpoint.Description}", new InputPrimitiveType(InputTypeKind.Uri, false), RequestLocation.Uri, null, null, null, InputOperationParameterKind.Client, true, false, false, false, true, false, false, null, null)
```

Comment:
```suggestion
                new InputParameter(KnownParameters.Endpoint.Name,KnownParameters.Endpoint.Name, $"{KnownParameters.Endpoint.Description}", new InputPrimitiveType(InputTypeKind.Uri, false), RequestLocation.Uri, null, null, null, InputOperationParameterKind.Client, true, false, false, false, true, false, false, null, null)
```

Reply:
It is not model property, we don't even use this `NameInRequest`.

Reply:
Even we does not use it, we may better to keep consistent.